centralstore/local_store.go

```diff
@@ -210,6 +218,8 @@ func (lrs *LocalStore) GetTrace(ctx context.Context, traceID string) (*CentralTr
 // after this call (although kept spans will have counts updated when late
 // spans arrive).
 func (lrs *LocalStore) GetStatusForTraces(ctx context.Context, traceIDs []string) ([]*CentralTraceStatus, error) {
+	_, span := otelutil.StartSpan(ctx, lrs.Tracer, "LocalStore.GetStatusForTraces")
+	defer span.End()
```

Comment:
Unrelated to this PR, but in the below for loop, since we know the exact length of `statuses` and that we'll be completely filling it, I think it is faster to do `statuses[i] = ` than `append`.

Reply:
Because I'm That Guy, I decided to try to verify this claim with a couple of benchmarks, and I think that the modern Go compiler is just as good in either case. These literally run to within half a nanosecond of each other. Since I prefer the append model in case of an early exit, I'd rather do that.

```go
type testStruct struct {
	foo string
	bar int
}

func BenchmarkAssign(b *testing.B) {
	data := make([]*testStruct, b.N)
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		item := &testStruct{foo: "hi", bar: 1}
		data[i] = item
	}
}

func BenchmarkAppend(b *testing.B) {
	data := make([]*testStruct, 0, b.N)
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		item := &testStruct{foo: "hi", bar: 1}
		data = append(data, item)
	}
}
```